Session timer enhancement

pcarana opened the pull request:
Under a load test scenario, the timer used for session expiration can affect the packet translation time. When the expired sessions are being removed, assuming that there are serveral expired sessions, the spin lock used can delay the packet translation the same time that the lock spends removing sessions.

The proposal (thanks @ydahhrk) is to "independize" the session removal timer, so that its behavior can be customized. The timer will do this:
1. Start execution every **2 seconds** and try to remove **1024** expired sessions.
   1. If this is enough, keep the same time and sessions to remove.
   1. If this isn't enough, the next execution will start in **1 second** and will try to remove **2048** sessions.
      1. If this is enough, the next execution will start in **2 seconds** and will try to remove **1024** sessions (aka begin from the start).
      1. If this isn't enough, the next execution will start in **0.5 seconds** and will try to remove **4096** sessions.
         1. _And so on..._

Removing expired sessions isn't as important as translating packets, but it's still something that must be done (we don't want to keep expired sessions).

Commit:
Divide timer into session and global timer

Due to a possible bottleneck scenario, the session timer needed to be
sepparated from the other timers in order to have control on the
execution time of such timer. The scenario is this: while the sessions
are being removed, the package translation could be delayed by the
execution of the sessions removal task. This is a massive trafic
scenario that could happen when there's a huge amount of expired
sessions to remove.

Commit:
Remove debug logs

ydahhrk reviewed (changes requested):
Thank you. Mainly please move the session timer to the bib table so a busy Jool instance will not affect the others.
It would be nice to see a few numbers out of a test as well.

Commit:
Use boolean labels instead of digits

Commit:
Delete session timer to relocate it

Commit:
Add a session timer to each BIB table

Now the timer is related to each table, therefore it doesn't affect
other instances.

The code is still a work in progress, some parameters at some functions
can be removed.

Commit:
Remove useless parameters

Commit:
Receive timer's initial config as parameter

Commit:
Fix wrong calls to "bib_create"

Commit:
Fix more broken tests

There are still some tests with warnings due to the "xlator" calls
when the session timers are removed; I've taken this as a general
warning: probably the BIB shouldn't be getting the "xlator".

I'm analizing what to do, since this is probably not the best solution
(to get the "xlator" from the BIB scope).

Commit:
Remove useless parameters, use constants instead

Commit:
Declare bib_clean as static function and relocate code to use it

Commit:
Destroy session timers when releasing BIB

Commit:
Remove reference from timer to DB, use the BIB table instead

pcarana commented on mod/stateful/session_timer.c:
```diff
@@ -0,0 +1,71 @@
+	u64 max_session_rm;
+};
+
+static struct clean_params sess_params;
```
Now the parameters and the timers belong to the `bib_table` structure.

pcarana commented on mod/stateful/session_timer.c:
```diff
@@ -0,0 +1,71 @@
+
+static void timer_function(unsigned long arg)
+{
+	xlator_foreach(clean_state, &sess_params);
```
This has been removed.

pcarana commented on mod/stateful/session_timer.c:
```diff
@@ -0,0 +1,71 @@
+	timer.function = timer_function;
+	timer.expires = 0;
+	timer.data = 0;
+	sess_params.pend_rm = 0;
```
Thanks, now the boolean labels are used.

pcarana reviewed (commented):
Requested changes had been made in 9e214c6

Commit:
Kill timers before the trees

Commit:
Merge with fake-nat64 measurement tool
